### PR TITLE
Search sources

### PR DESCRIPTION
**Pull request by bbfsdev**

### CODE REVIEW

File: src/helpers/consts.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ export const CT_KITEI_MAKOR           = 'KITEI_MAKOR';
 export const CT_PUBLICATION           = 'PUBLICATION';
 export const CT_LELO_MIKUD            = 'LELO_MIKUD';
 
+// Source content type
+export const CT_SRC          = 'SOURCE';
```
Comment (edoshor) [inline src/helpers/consts.js]: What is this ?
content type is for units and collections. not sources.

File: src/redux/modules/mdb.js [hunk 2]
```diff
@@ -76,21 +78,25 @@ export const actions = {
 
   receiveCollections,
   receiveContentUnits,
+  receiveSources,
```
Comment (edoshor) [inline src/redux/modules/mdb.js]: Anything related to sources data should live in the 'sources' redux module.
'mdb' redux module is for collections and units.

Please move all this things to sources module and try to reuse code as much as possible.

File: src/components/Search/SearchResults.js [hunk 3]
```diff
@@ -4,20 +4,31 @@ import PropTypes from 'prop-types';
...
+  SEARCH_I_FILTER_NAMES,
+  SEARCH_I_NAMES,
+  SEARCH_I_SOURCE,
+  SEARCH_I_TOPIC,
+} from '../../helpers/consts';
 
 class SearchResults extends Component {
   static propTypes = {
     results: PropTypes.object,
+    getSourcePath: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/components/Search/SearchResults.js]: Done.

File: src/components/Search/SearchResults.js [hunk 4]
```diff
@@ -163,38 +176,220 @@ class SearchResults extends Component {
...
+    });
+  };
+
   render() {
-    const { filters, wip, err, results, pageNo, pageSize, language, t, handlePageChange, cMap, cuMap, location } = this.props;
+    const { filters, wip, err, queryResult, getSourcePath, pageNo, pageSize, language, t, handlePageChange, location } = this.props;
+    const { search_result: results, intents = []} = queryResult;
 
-    const wipErr = WipErr({ wip, err, t });
+    const wipErr = WipErr({ wip: wip || getSourcePath === undefined, err, t });
```
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/components/Search/SearchResults.js]: Sent you example in whatsapp, any ideas?

File: src/helpers/consts.js (hunk 1 above)
Comment (bbfsdev) [reply src/helpers/consts.js]: It is used in canonicalLink in link.js:
 64   // source
 65   if (entity.content_type === CT_SRC){
 66     return `/sources/${entity.id}`;
 67   }

Want to call it differently?

File: src/components/Search/SearchResults.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/components/Search/SearchResults.js]: What you wrote is not correct.
Added sourcesLoaded boolean from redux. Using it in WIP.
After sources are loaded this param is set correctly.

File: src/components/Search/SearchResults.js [hunk 5]
```diff
@@ -82,7 +94,7 @@ class SearchResults extends Component {
 
     let filmDate = '';
     if (cu.film_date) {
-      filmDate = t('values.date', { date: cu.film_date });
+      filmDate = t('values.date', { date: new Date(cu.film_date) });
```
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/components/Search/SearchResults.js]: Not clear please explain.

File: src/components/Search/SearchResults.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (bbfsdev) [reply src/components/Search/SearchResults.js]: Fixed.

File: src/redux/modules/mdb.js (hunk 2 above)
Comment (bbfsdev) [reply src/redux/modules/mdb.js]: Fixed.

**Pull request merged**